build-tools/Xamarin.Android.Tools.BootstrapTasks/Xamarin.Android.Tools.BootstrapTasks/CheckApiCompatibility.cs

```diff
@@ -0,0 +1,294 @@
+					return !Log.HasLoggedErrors;
+				}
+
+				// Before validate, check that zip files were decompressed.
+				var zipFiles = Directory.GetFiles (referenceContractPath, "*.zip");
+				foreach (var zipFile in zipFiles) {
+					using (var zip = ZipArchive.Open (zipFile, FileMode.Open)) {
+						zip.ExtractAll (referenceContractPath);
+					}
+				}
+
+				// When comparing the current changes assembly with the reference assembly we should not report warning as error.
+				// Here is why: The Acceptable-Breakages file contains the issues that are accepted between versions. i.e.: version 1 against version 2
+				// At this stage we are comparing V2 against V2 (reference), so the issues listed on the file will probably don't happen, so we should
+				// not report that the file has issues listed that are not happening.
+				ValidateApiCompat (referenceContractPath, false);
+			}
+
+			return !Log.HasLoggedErrors;
+		}
+
+		// Validates Api compatibility between contract (previous version) and implementation (current version)
+		// We do that by using Microsoft.DotNet.ApiCompat.dll
+		void ValidateApiCompat (string contractPath, bool reportWarningAsError)
+		{
+			const string ApiCompatTemp = "ApiCompatTemp";
+
+			var apiCompat = Path.Combine (ApiCompatPath, "Microsoft.DotNet.ApiCompat.exe");
+			var contractPathDirectory = Path.Combine (contractPath, ApiCompatTemp);
+			var targetImplementationPathDirectory = Path.Combine (TargetImplementationPath, ApiCompatTemp);
+
+			try {
+				// Copy interesting assemblies to a temp folder.
+				// This is done to avoids the Microsoft.DotNet.ApiCompat.exe to analyze unwanted assemblies
+				// We need to validate assembly exist in both contract and implementation folders.
+				Directory.CreateDirectory (contractPathDirectory);
+				Directory.CreateDirectory (targetImplementationPathDirectory);
+
+				foreach (var assemblyToValidate in assemblies) {
+					var contractAssembly = Path.Combine (contractPath, assemblyToValidate);
+					if (!File.Exists (contractAssembly)) {
+						Log.LogMessage ($"Contract assembly {assemblyToValidate} does not exists in the contract path.");
+						continue;
+					}
+
+					var implementationAssembly = Path.Combine (TargetImplementationPath, assemblyToValidate);
+					if (!File.Exists (implementationAssembly)) {
+						Log.LogError ($"Implementation assembly {assemblyToValidate} exists in the contract path but not on the implementation folder.");
+						return;
+					}
+
+					File.Copy (contractAssembly, Path.Combine (contractPathDirectory, assemblyToValidate), true);
+					File.Copy (implementationAssembly, Path.Combine (targetImplementationPathDirectory, assemblyToValidate), true);
+				}
+
+				using (var genApiProcess = new Process ()) {
+
+					genApiProcess.StartInfo.FileName = apiCompat;
+					genApiProcess.StartInfo.Arguments = $"{contractPathDirectory} -i {targetImplementationPathDirectory} ";
```

Comment:
For "sanity", directories & path names should be quoted with `"` in case they contain spaces.

Reply:
Sure, Fixed!